lib/options.nix

```diff
@@ -93,16 +93,24 @@ rec {
           visible = opt.visible or true;
           readOnly = opt.readOnly or false;
           type = opt.type.name or null;
-        }
-        // (if opt ? example then { example = scrubOptionValue opt.example; } else {})
-        // (if opt ? default then { default = scrubOptionValue opt.default; } else {})
-        // (if opt ? defaultText then { default = opt.defaultText; } else {});
+        };
+        docOptionDefault =
+          if opt ? defaultText then { default = opt.defaultText; }
+          else if opt ? default then { default = scrubOptionValue opt.default; }
+          else { };
+        docOptionExample =
+          if opt ? example then { example = scrubOptionValue opt.example; }
+          else if opt.type == lib.types.bool then
+            { example = !docOptionDefault.default or false; }
+          else if opt ? type && isBool (docOptionDefault.default or null) && opt.type.check (!docOptionDefault.default) then
```

Comment:
That should never happen since it's only evaluated if `isBool (docOptionDefault.default or null)`